### ISSUE TITLE
Multimesh splitting

### ISSUE THREAD

**Issue by Norodix** [HungryProton/scatter#79]
We are currently working on a game with a relatively large map and wanted to improve the performance of the foliage.
I implemented a small addition to the plugin that makes it possible to automatically split up the generated multimesh into many smaller chunks. This improves performance on larger maps because chunks can be culled one by one.

The approach currently is to create sibling multimesh instances and copy the instances to those based on their position (AABB).
The original generated multimesh is preserved and hidden.

Its quite simple and you can see the implementation on my fork.
https://github.com/Norodix/scatter

![image](https://user-images.githubusercontent.com/36245389/184479726-ce104025-b085-4c72-9608-421581418179.png)

My question is whether you think this approach is suitable for the plugin as a whole.
Should I clean up the code, fix bugs and create a pull request?

**Comment by HungryProton** [comment]
That's great! It's a feature I've been asked a lot before, but never got the time to work on it. If you create a PR, I'll gladly merge it.

There's room for a few optimizations in the code, but that's not critical. Like at line 282, why not simply use `mmi.get_aabb()` directly instead of looping through every instance to recalculate it ? There's also probably a way to avoid looping through everything 3 times when creating the chunks, but again, I haven't profiled anything yet, so it may not even be an issue.

**Comment by Norodix** [comment]
I noticed some of those too. Just wanted to get something working so we can discuss whether this split sibling generation is a good approach in your opinion. Im glad you like it.

I will improve the performance, then create a pull request!

Anything else before it can be merged? Some coding style issue, or something like that?

**Comment by HungryProton** [comment]
For the coding style, I'm using these guidelines: https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_styleguide.html

Otherwise, should be good! :+1:

**Comment by Norodix** [comment]
I did some optimizations and as expected it made everything much worse.

I experimented with some solutions and it turns out adding many nodes to the scene tree is very slow. Not even linearly, if I added 3000 it was around 3 seconds, 6000 was around half a minute. I didnt want to go higher. Accidentally my previous code sort of solved this by discarding empty multimesh instances.

I reorganized it so the new one does the same. Not its quite nice and fast.
